### PR TITLE
Improvements to custom library + oot driver class + documentation

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

File: include/app/lib/custom.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * @defgroup app_lib_custom Custom Application Library
+ * @{
+ * @brief An example of an application-level library.
+ *
+ * This library illustrates how create application-level libraries. Splitting
+ * code in libraries enables code reuse and modularity, also easing testing.
+ *
+ *
```
[the remark this answers is not in the document]
Comment (gmarull) [reply include/app/lib/custom.h]: done

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          cd doc
+          doxygen
+          # required if github pages are not built using Jekyll
+          touch _build_doxygen/html/.nojekyll
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4
+        if: github.ref == 'refs/heads/main'
+        with:
+          branch: gh-pages
```
[the remark this answers is not in the document]
Comment (gmarull) [reply .github/workflows/docs.yml]: good stuff, updated

File: doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+This is a simple Sphinx documentation setup for ``example-application``. You can
+reference Zephyr documentation items, like
+:external+zephyr:ref:`getting_started` or :external+zephyr:c:func:`k_sleep`.
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
```
Comment (fabiobaltieri) [inline doc/index.rst:21]: this does not seem to link correctly from the artifact generated page, points to `http://localhost:8000/sphinx/py-modindex.html` which is not there

```console
~/x$ ls sphinx/
_sources/  genindex.html  objects.inv  searchindex.js
_static/   index.html     search.html
```

Comment (gmarull) [reply doc/index.rst]: removed modindex entry, there are no Python modules to be listed

**Review by fabiobaltieri — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by gmarull** [comment]
last push: fixed typo in readme

**Comment by fabiobaltieri** [comment]
@kartben can you take a look as well?

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
Artifact can be downloaded for now, and it looks like:

image

**Comment by wbober** [comment]
Can we put a section with the doxygen API from the sphinx docs please?

**Comment by gmarull** [comment]
last push: some tweaks, plus added a custom driver class. All can be run on nrf52840dk (using custom_plank target) and nucleo_f302r8 boards.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
Talked to @kartben, agreed to move on with it.

**Comment by fabiobaltieri** [comment]
This now publishes at https://zephyrproject-rtos.github.io/example-application/sphinx/ and https://zephyrproject-rtos.github.io/example-application/doxygen/, @gmarull you may want to add a static index for https://zephyrproject-rtos.github.io/example-application

**Comment by gmarull** [comment]
> This now publishes at https://zephyrproject-rtos.github.io/example-application/sphinx/ and https://zephyrproject-rtos.github.io/example-application/doxygen/, @gmarull you may want to add a static index for https://zephyrproject-rtos.github.io/example-application

sphinx is now the root doc

**Pull request merged**